client/js/context/UserProvider.js

```diff
@@ -0,0 +1,46 @@
+import { Query } from "react-apollo";
+import React from "react";
+import PropTypes from "prop-types";
+
+export const UserContext = React.createContext();
+
+export const UserProvider = ({ children }) => {
+  return (
+    <Query query={USER_QUERY}>
+      {({ data, error, loading }) => {
+        const user = data && data.user ? data.user : null;
+        return (
+          <UserContext.Provider value={{ user, loading }}>
+            {children}
+          </UserContext.Provider>
+        );
+      }}
+    </Query>
+  );
+};
+
+ViewerProvider.propTypes = {
+  children: PropTypes.object.isRequired
+};
+
+const USER_QUERY = gql`
+  query User($id: ID) {
+    User(id: "cjwtisae10ech0103e84u962r") {
```

Comment:
Is this a `secret`? Move to a .env and use react-native-dotenv. Squash the commit so this key is not available publicly

Reply:
This is the id of a sample user. It will be substituted by the current user id once context is implemented.